test/LaunchDarkly.EventSource.Tests/Stubs/StubHandler.cs

```diff
@@ -26,41 +28,154 @@ public class StubMessageHandler : HttpMessageHandler
             if (_responses.Count == 0)
                 throw new InvalidOperationException("No response configured");
```

Comment:
This is a minor issue but I noticed that this logic doesn't exactly work - if it gets a request and there's no queued response, it will throw an exception, but that exception is just treated as a generic HTTP error. It would be nice if it could actually terminate the current unit test somehow, but I can't think how.

Reply:
If this is just an issue with the stub and is entirely encapsulated in the tests, could we just put an assert inside of the `SendAsync` stub?

Reply:
To be clear, the issue is that if the stub server receives 2 HTTP requests, but we had only set it up with 1 response, then something is wrong with the test and it should probably die. But I don't think using `Assert` here would do anything more than `throw` is doing. An assertion failure just causes an exception to be thrown, that's how it's implemented.